src/SpaFronted/Parsobober.DesignExtractor/ProgramContextExtractor.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Parsobober.Pkb.Ast;
+using Parsobober.Pkb.Relations.Abstractions.Creators;
+
+namespace Parsobober.Simple.Parser.Extractor
+{
+    internal class ProgramContextExtractor(IProgramContextCreator creator) : SimpleExtractor
+    {
+        private readonly IProgramContextCreator creator = creator;
```

Comment:
you have 2 same private props `creator` one from primary constructors and second one you wrote